exercises/practice/connect/.meta/Example.roc

```diff
@@ -0,0 +1,102 @@
+module [winner]
+
+Cell : [StoneO, StoneX, Empty]
+Board : List (List Cell)
+Position : { x : U64, y : U64 }
+
+## Parse a string to a Board
+parse : Str -> Result Board [InvalidCharacter U8]
+parse = \boardStr ->
+    boardStr
+    |> Str.trim
+    |> Str.split "\n"
+    |> List.mapTry \row ->
+        row
+        |> Str.toUtf8
+        |> List.dropIf \char -> char == ' '
+        |> List.mapTry \char ->
+            when char is
+                'O' -> Ok StoneO
+                'X' -> Ok StoneX
+                '.' -> Ok Empty
+                _ -> Err (InvalidCharacter char)
+
+## Ensure that the board has a least one cell, and that all rows have the same length
+validate : Board -> Result Board [InvalidBoardShape]
+validate = \board ->
+    rowLengths = board |> List.map List.len |> Set.fromList
+    if Set.len rowLengths != 1 || rowLengths == Set.fromList [0] then
+        Err InvalidBoardShape
+    else
+        Ok board
+
+winner : Str -> Result [PlayerO, PlayerX] [NotFinished, InvalidCharacter U8, InvalidBoardShape]
+winner = \boardStr ->
+    maybeBoard = boardStr |> parse?
+    board = maybeBoard |> validate?
```

Comment:
You could update validate to return a `Result {} [InvalidBoardShape]` and then do this instead.
```suggestion
    board = parse? boardStr
    validate? maybeBoard
```

Reply:
Nice suggestion, thanks.